File: models/resultado.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Resultado extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      // Un resultado pertenece a un partido
      Resultado.belongsTo(models.Partido, { foreignKey: 'partido_id' });
    }
  }
  Resultado.init({
    partido_id: DataTypes.INTEGER,
    goles_local: DataTypes.INTEGER,
    goles_visitante: DataTypes.INTEGER
  }, {
    sequelize,
    modelName: 'Resultado',
    timestamps: true
  });
  return Resultado;
};

File: Flutter/src/nemer_app/lib/models/resultado.dart
class Resultado {
  final int? id;
  final int partidoId;
  final int golesLocal;
  final int golesVisitante;

  Resultado({
    this.id,
    required this.partidoId,
    required this.golesLocal,
    required this.golesVisitante,
  });

  factory Resultado.fromJson(Map<String, dynamic> json) {
    return Resultado(
      id: json['id'],
      partidoId: json['partido_id'] ?? json['partidoId'] ?? 0,
      golesLocal: json['goles_local'] ?? json['golesLocal'] ?? 0,
      golesVisitante: json['goles_visitante'] ?? json['golesVisitante'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "partido_id": partidoId,
      "goles_local": golesLocal,
      "goles_visitante": golesVisitante,
    };
  }
}

